### ISSUE TITLE
consistent-function-scoping should ignore jest mocks

### ISSUE THREAD

**Issue by BridgeAR** [sindresorhus/eslint-plugin-unicorn#805]
Using `jest.mock()` does not work properly with `consistent-function-scoping` as it's not allowed to reference any out of scope variables as mocked function.

```ReferenceError: /__tests__/search.spec.tsx: The module factory of `jest.mock()` is not allowed to reference any out-of-scope variables.``` 

```js
jest.mock("next/dynamic", () => () => "dynamic-drawer");
```
